### DISCUSSION TITLE
Refactor logging

### DISCUSSION THREAD

**Discussion by xQwexx** [GrapesJS/grapesjs#3941] [Ideas]
So currently the logging calls happen in the editor model class, which in not the best architecture decision,
because some classes need acquire the editor model just for the logging purposes which could have done whit an include.
So I proposing to move these functions to a separate logging class in the utils dir.
I can make the pr just first I wanted to discuss this.
```
  log(msg, opts = {}) {
    const { ns, level = 'debug' } = opts;
    this.trigger('log', msg, opts);
    level && this.trigger(`log:${level}`, msg, opts);

    if (ns) {
      const logNs = `log-${ns}`;
      this.trigger(logNs, msg, opts);
      level && this.trigger(`${logNs}:${level}`, msg, opts);
    }
  },

  logInfo(msg, opts) {
    this.log(msg, { ...opts, level: 'info' });
  },

  logWarning(msg, opts) {
    this.log(msg, { ...opts, level: 'warning' });
  },

  logError(msg, opts) {
    this.log(msg, { ...opts, level: 'error' });
  },
```

**Comment by artf** [comment]
Well, yeah it sounds correct but the problem here is `this.trigger('log', msg, opts)`. We still need the EditorModel reference in order to trigger the log on the editor level. This mechanism allows listening events on each editor separately
```js
const editor = grapesjs.init(...);
editor.on('log:error', ...)
```
